engine/src/main/java/com/datatorrent/stram/engine/Node.java

```diff
@@ -485,6 +485,10 @@ protected void deactivateSinks()
 
   void checkpoint(long windowId)
   {
+    if (operator instanceof Operator.CheckpointNotificationListener) {
```

Comment:
@PramodSSImmaneni looks like majority is in favor of making the change. Can you please squash the commits along with it?